### PR TITLE
feat: support gleam

### PR DESCRIPTION
**Pull request by wesleimp**

### CODE REVIEW

File: lua/lspconfig/server_configurations/gleam.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = {'gleam', 'lsp'},
+    filetypes = { 'gleam' },
+    root_dir = function(fname)
+      return util.root_pattern('gleam.toml', '.git')(fname) or vim.loop.os_homedir()
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/gleam.lua:8]: remove the homedir fallback, and the pattern should reduce to `root_dir = util.root_pattern('gleam.tom', '.git')`

File: lua/lspconfig/server_configurations/gleam.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    end,
+  },
+  docs = {
+    description = [[
+https://github.com/gleam-lang/gleam
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/gleam.lua:13]: Should include some mention of installation

File: lua/lspconfig/server_configurations/gleam.lua [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+```lua
+require('lspconfig').gleam.setup({})
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/gleam.lua:16]: This is redundant, and added automatically in docgen.

File: lua/lspconfig/server_configurations/gleam.lua [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+```
+]],
+    default_config = {
+      cmd = { 'gleam', 'lsp' },
+      root_dir = [[root_pattern("gleam.toml", ".git") or vim.loop.os_homedir()]],
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/gleam.lua:21]: Update this when you fix the root pattern

**Comment by lpil** [comment]
The Gleam LSP has now been released, time to move forward with this PR? :)

**Comment by justinmk** [comment]
Waiting for followup.

**Pull request closed without merging**